packages/core/src/interfaces.ts

```diff
@@ -0,0 +1,58 @@
+import { BigNumber, ethers } from 'ethers'
+
+export interface IEpochTreeLeaf {
+    epochKey: BigInt
+    hashchainResult: BigInt
+}
+
+export interface ISettings {
+    readonly globalStateTreeDepth: number
+    readonly userStateTreeDepth: number
+    readonly epochTreeDepth: number
+    readonly attestingFee: ethers.BigNumber
+    readonly epochLength: number
+    readonly numEpochKeyNoncePerEpoch: number
+    readonly maxReputationBudget: number
+}
+
+export interface IUnirepState {
+    readonly settings: ISettings
+    currentEpoch: number
+    latestProcessedBlock: number
+    GSTLeaves: { [key: string]: string[] }
+    epochTreeLeaves: { [key: string]: string[] }
+    latestEpochKeyToAttestationsMap: { [key: string]: string[] }
+    nullifiers: string[]
+}
+
+export interface IUserStateLeaf {
+    attesterId: BigInt
+    reputation: IReputation
+}
+
+export interface IReputation {
```

Comment:
I also see we use same Interface in circuits/test/utils.
https://github.com/Unirep/Unirep/blob/fix/json-serialization/packages/circuits/test/utils.ts#L34

Reply:
`IAttestation` is different from `IReputation`